javascript/javascript3/week2/homework.js

```diff
@@ -0,0 +1,108 @@
+// fetch movies json
+////////////////////////////////////////////
+fetch('https://gist.githubusercontent.com/pankaj28843/08f397fcea7c760a99206bcb0ae8d0a4/raw/02d8bc9ec9a73e463b13c44df77a87255def5ab9/movies.json')
+    .then(response => response.json())
+    .then(moviesData => {
+        const badMovies = moviesData.filter(movie => movie.rating < 4)
+        const recentBadMovies = badMovies.filter(movie => movie.year >= 2000)
+    })
+
+// Promise that resolves after set time
+////////////////////////////////////////////
+function returnPromise(resolveAfter) {
+    return new Promise(resolve => {
+        setTimeout(() => {
+            resolve();
+        }, resolveAfter * 1000);
+    });
+}
+    
+returnPromise(8).then(() => {
+    console.log("I am timeout resolve"); // logged out after 8 seconds
+});
+  
+// use it with async/await
+////////////////////////////////////////////
+async function getPromise() {
+    try {
+        const asyncResponse = await returnPromise(8).then(() => {
+            console.log("I am called asynchronously"); // logged out after 8 seconds
+        });
+        return asyncResponse;
```

Comment:
This solution will absolutely work, but it seems like a little bit of mix up between the use of async/await and promises. In production code, you would want to use either async/await syntax or promise syntax in order to stay more consistent :) 
In this case you could use something like:
```suggestion
        await returnPromise(8)
        console.log("I am called asynchronously"); // logged out after 8 seconds
```
because you are awaiting, the print statement will not run untill returnPromise has resolved (or rejected).